From 610c0f73d42b19028dc4a7a64f6d50d953fb581d Mon Sep 17 00:00:00 2001
From: Ramana Radhakrishnan <ramana.radhakrishnan@arm.com>
Date: Thu, 22 May 2014 16:07:20 +0100
Subject: [PATCH] Include asm/ptrace.h in aarch64-linux-nat.c

A recent change to glibc removed asm/ptrace.h from user.h for
AArch64. This meant that cross-native builds of gdb using trunk
glibc broke because aarch64-linux-nat.c because user_hwdebug_state
couldn't be found.

Fixed by including asm/ptrace.h like other ports.

2014-05-22  Ramana Radhakrishnan  <ramana.radhakrishnan@arm.com>

       * aarch64-linux-nat.c (asm/ptrace.h): Include.
---
 gdb/ChangeLog           | 4 ++++
 gdb/aarch64-linux-nat.c | 1 +
 2 files changed, 5 insertions(+)

diff --git a/gdb/ChangeLog b/gdb/ChangeLog
index 12aa42435f..a506e0997b 100644
--- a/gdb/ChangeLog
+++ b/gdb/ChangeLog
@@ -1,3 +1,7 @@
+2014-05-22  Ramana Radhakrishnan  <ramana.radhakrishnan@arm.com>
+
+	* aarch64-linux-nat.c (asm/ptrace.h): Include.
+
 2013-11-19  Joel Brobecker  <brobecker@adacore.com>
 
 	* version.in: Set GDB version number to 7.6.2.
diff --git a/gdb/aarch64-linux-nat.c b/gdb/aarch64-linux-nat.c
index f68561944d..33f192585a 100644
--- a/gdb/aarch64-linux-nat.c
+++ b/gdb/aarch64-linux-nat.c
@@ -33,6 +33,7 @@
 
 #include <sys/ptrace.h>
 #include <sys/utsname.h>
+#include <asm/ptrace.h>
 
 #include "gregset.h"
 
-- 
2.17.1

